### ISSUE TITLE
Setting the Value of a Dropdown is Inefficient

### ISSUE THREAD

**Issue by BeksOmega** [google/blockly#2493]
### Problem statement

validateOptions_, doClassValidation_, and doValueUpdate_ all loop through the dropdown's options. This is inefficient.

### Proposal

validateOptions_ should create a map of Language-Neutral keys to Human-Readable options. That way it could be queried when selecting a value, instead of having to loop through all of the options.

### Additional Information

If this change is made option trimming (prefix/suffix matching) would need to be reworked so that the map contains the trimmed options.

**Comment by samelhusseini** [comment]
Yeah, I agree, it doesn't make sense that we constantly regenerate the options list, especially that the returned values could be different. 
I think it makes sense to re-generate the options list when the dropdown is opened.

**Comment by BeksOmega** [comment]
Relevant Discussion: https://github.com/google/blockly/pull/3036#discussion_r325908979
